src/components/organisms/BannerAdmin/BannerAdmin.tsx

```diff
@@ -1,90 +1,176 @@
-    (msg?: string) => {
+    (banner?: Banner) => {
       if (!venueId) return;
-
-      makeUpdateBanner(venueId, (errorMsg) => setError(errorMsg))(msg);
+      makeUpdateBanner({ venueId, banner });
+      onClose();
     },
-    [venueId]
+    [venueId, onClose]
   );
 
-  const saveBanner = useCallback(
-    (e: React.MouseEvent<HTMLButtonElement>) => {
-      e.preventDefault();
-
-      if (!inputFieldRef.current) return;
-
-      updateBannerInFirestore(inputFieldRef.current.value);
+  const onSubmit = useCallback(
+    (data: Banner) => {
+      showModal();
+      setBannerDate(data);
     },
-    [updateBannerInFirestore]
+    [showModal]
   );
 
-  const clearBanner = useCallback(() => updateBannerInFirestore(""), [
-    updateBannerInFirestore,
-  ]);
+  const clearBanner = useCallback(() => {
+    updateBannerInFirestore(undefined);
+    reset();
+  }, [updateBannerInFirestore, reset]);
+
+  const confirmChangeBannerData = useCallback(() => {
+    updateBannerInFirestore(bannerData);
+    hideModal();
+  }, [bannerData, updateBannerInFirestore, hideModal]);
 
   return (
-    <div className="container">
-      <div className="row">
-        <div className="col">
-          <form>
-            <div className="form-group">
-              <label htmlFor="bannerMessage">Banner Message</label>
-
-              <input
-                ref={inputFieldRef}
-                type="text"
-                defaultValue={existingBannerMessage}
-                onChange={handleInputChange}
-                placeholder="Enter banner message here..."
-                autoComplete="off"
-              />
-
-              {error && <span className="error">{error}</span>}
-            </div>
-
-            <div className="form-inline justify-content-between">
-              <button
-                className="btn btn-danger"
-                type="reset"
-                onClick={clearBanner}
-              >
-                Clear Banner
-              </button>
-
-              <button
-                className="btn btn-primary"
-                type="submit"
-                onClick={saveBanner}
-              >
-                Save Banner
-              </button>
-            </div>
-          </form>
+    <div className="BannerAdmin">
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <InputField
+          ref={register}
+          name="title"
+          placeholder="Please type your announcement title. (optional)"
+          defaultValue={venue?.banner?.title}
+          containerClassName="BannerAdmin__input-container"
+          inputClassName="BannerAdmin__input-text"
+          autoComplete="off"
+        />
+
+        <div className="form-group">
+          <textarea
+            name="content"
+            defaultValue={venue?.banner?.content}
+            className="BannerAdmin__input-text"
+            placeholder="Please type your announcement content"
+            ref={register({ required: true })}
```

Comment:
Why are some `ref`'s here at the top of the props, and others at the bottom. Please pick one and be consistent. I think top of props may be more common in the codebase off the top of my head.

Reply:
fixed 771b70a